File: 1.0/src/c/cz/mg/vulkan/PFN_vkCmdSetDiscardRectangleEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkCmdSetDiscardRectangleEXT_callNative(JNIEnv* env, jclass jc, jlong address, jlong commandBuffer, jint firstDiscardRectangle, jint discardRectangleCount, jlong pDiscardRectangles){
    (void)env;
    (void)jc;
    PFN_vkCmdSetDiscardRectangleEXT f = (PFN_vkCmdSetDiscardRectangleEXT)jniLongToFunctionPointer(address);
    f(
        *((VkCommandBuffer*)jniLongToPointer(commandBuffer)),
        (uint32_t)firstDiscardRectangle,
        (uint32_t)discardRectangleCount,
        ((VkRect2D*)jniLongToPointer(pDiscardRectangles))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkCmdSetDiscardRectangleEXT.java
package cz.mg.vulkan;

public class PFNvkCmdSetDiscardRectangleEXT extends VkFunctionPointer {
    public PFNvkCmdSetDiscardRectangleEXT() {
    }

    protected PFNvkCmdSetDiscardRectangleEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkCmdSetDiscardRectangleEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkCmdSetDiscardRectangleEXT(long value) {
        setValue(value);
    }

    public PFNvkCmdSetDiscardRectangleEXT(VkInstance instance) {
        super(instance, new VkString("vkCmdSetDiscardRectangleEXT"));
    }

    public void call(VkCommandBuffer commandBuffer, int firstDiscardRectangle, int discardRectangleCount, VkRect2D pDiscardRectangles){
        callNative(getValue(), commandBuffer != null ? commandBuffer.getVkAddress() : VkPointer.getNullAddressNative(), firstDiscardRectangle, discardRectangleCount, pDiscardRectangles != null ? pDiscardRectangles.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long commandBuffer, int firstDiscardRectangle, int discardRectangleCount, long pDiscardRectangles);

}
